src/features/Dashboard/components/TokenPanel/AddToken.tsx

```diff
@@ -78,12 +78,16 @@ export function AddToken(props: Props) {
     return isValid;
   };
 
-  const handleAddTokenClick = () => {
+  const handleAddTokenClick = async () => {
     if (!validateForm()) {
       return;
     }
-
+    const { coinGeckoId } = await CustomAssetService.instance.fetchCoingeckoID(
```

Comment:
Can we add a loading state while this is fetched? Disabling the cancel button and showing the loading spinner on the add token button?